xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp

```diff
@@ -945,6 +945,51 @@ std::string CVideoPlayerVideo::GetPlayerInfo()
   else
     s << ", pc:none";
 
+  if (m_picture.hasDisplayMetadata && m_picture.displayMetadata.has_primaries &&
+      m_picture.displayMetadata.display_primaries[0][0].num)
+  {
+    double prim[3][2];
+    double wp[2];
+
+    for (int i = 0; i < 3; i++)
+    {
+      for (int j = 0; j < 2; j++)
+        prim[i][j] = static_cast<double>(m_picture.displayMetadata.display_primaries[i][j].num) /
+                     static_cast<double>(m_picture.displayMetadata.display_primaries[i][j].den);
+    }
+
+    for (int j = 0; j < 2; j++)
+      wp[j] = static_cast<double>(m_picture.displayMetadata.white_point[j].num) /
+              static_cast<double>(m_picture.displayMetadata.white_point[j].den);
+
+    s << "|Primaries: Red:" << std::fixed << std::setprecision(3) << prim[0][0];
+    s << "," << std::fixed << std::setprecision(3) << prim[0][1];
+    s << ", Green:" << std::fixed << std::setprecision(3) << prim[1][0];
+    s << "," << std::fixed << std::setprecision(3) << prim[1][1];
+    s << ", Blue:" << std::fixed << std::setprecision(3) << prim[2][0];
+    s << "," << std::fixed << std::setprecision(3) << prim[2][1];
+    s << ", WP:" << std::fixed << std::setprecision(3) << wp[0];
+    s << "," << std::fixed << std::setprecision(3) << wp[1];
```

Comment:
Thanks for including a screenshot. Looking it at, I think a little spacing would help this debug line. A space after each `:`, and maybe after each `,` too. And you could change/add punctuation to better represent the hierarcy of text. Here's a possibility:

```
Primaries - Red: 0.680, 0.320; Green: 0.265, 0.690; Blue: 0.150, 0.060; WP: 0.313, 0.329
```

Reply:
Or maybe in this other way it's clearer and more compact at the same time, the problem with the extra spaces and commas is that it becomes a very long line and in some resolutions it may not fit.

`Primaries: R(0.680 0.320), G(0.265 0.690), B(0.150 0.060),  WP(0.313 0.329)`

Reply:
nice decision